scripts: tests: sanitycheck: Added testcases for TestCase class

yerabolu opened the pull request.

aasthagr commented:
These two testcases are failing for me. Please check.

FAILED test_testinstance.py::test_subcases[testcases/tests/test_a-Found a test that does not start with test_-expected_subcases1] - AssertionError: assert ['1a', '1c', ...
FAILED test_testsuite_class.py::test_get_all_testcases - AssertionError: assert 19 == 7

yerabolu commented:
> These two testcases are failing for me. Please check.
> 
> FAILED test_testinstance.py::test_subcases[testcases/tests/test_a-Found a test that does not start with test_-expected_subcases1] - AssertionError: assert ['1a', '1c', ...
> FAILED test_testsuite_class.py::test_get_all_testcases - AssertionError: assert 19 == 7


Above tests are fixed and running.

yerabolu committed:
scripts: tests: sanitycheck: changes to sanitylib to sort the warnings

While testing the warnings in scan_file() format is not the same
everytime we run tests
Added sorted with reverse flag so that warnings are same for
every run

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

yerabolu committed:
scripts: tests: sanitycheck: Add testcases for TestCase class

test_testinstance.py: Add testcases to scan file and path for
sub testcases
test_data/testcases: Also added the ztest test files
test_testsuite_class.py: changed get_all_tests() to match
count of sub testcases in ztest files

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>
